quarkus-test-openshift/src/main/java/io/quarkus/test/openshift/utils/OpenShiftPropertiesUtils.java

```diff
@@ -0,0 +1,184 @@
+        return propertyValue.startsWith(ANNOTATED_CONFIG_MAP_PREFIX);
+    }
+
+    public static boolean isAnnotation(String propertyValue) {
+        return propertyValue.startsWith(ANNOTATION_PREFIX);
+    }
+
+    public static String createAnnotationPropertyKey() {
+        return createRandomPropertyKeyWithPrefix("annotation-");
+    }
+
+    public static String createSecretPropertyKey() {
+        return createRandomPropertyKeyWithPrefix("secret-");
+    }
+
+    public static String createConfigMapPropertyKey() {
+        return createRandomPropertyKeyWithPrefix("configmap-");
+    }
+
+    public static String createAnnotatedConfigMapPropertyKey() {
+        return createRandomPropertyKeyWithPrefix("annotated-configmap-");
+    }
+
+    public static PropertyToValue getMountConfigMap(String propertyValue) {
+        return getPropertyToValue(propertyValue, MOUNT_CONFIGMAP_PREFIX, MOUNT_CONFIGMAP_SEPARATOR, "configmap");
+    }
+
+    public static PropertyToValue getMountSecret(String propertyValue) {
+        return getPropertyToValue(propertyValue, MOUNT_SECRET_PREFIX, MOUNT_SECRET_SEPARATOR, "secret");
+    }
+
+    public static PropertyToValue getServiceAnnotation(String propertyValue) {
+        return getPropertyToValue(propertyValue, ANNOTATION_PREFIX, ANNOTATION_SEPARATOR, "annotation");
+    }
+
+    public static PropertyToObj getAnnotatedConfigMap(String propertyValue) {
+        var configMapNameToAnnotation = getPropertyToValue(propertyValue, ANNOTATED_CONFIG_MAP_PREFIX,
+                ANNOTATED_CONFIG_MAP_SEPARATOR, "annotated-configmap");
+        var annotationKeyToValue = getPropertyToValue(configMapNameToAnnotation.value(), "",
+                ANNOTATED_CONFIG_ANNOTATION_SEPARATOR, "annotation-configmap-annotation");
+        return new PropertyToObj(configMapNameToAnnotation.key(), annotationKeyToValue);
+    }
+
+    public static boolean checkPodReadinessWithStatusInsteadOfRoute(ServiceContext ctx) {
+        // using HTTPS & internal service URL & no exposed route => need to trust readiness
+        return isServingCertificateScenario(ctx) && ServingCertificateConfig.get(ctx).addServiceCertificate();
+    }
+
+    public static int getInternalHttpsPort(ServiceContext ctx) {
+        return ctx.getOwner().getProperty(QUARKUS_HTTPS_PORT_PROPERTY)
+                .filter(StringUtils::isNotBlank)
+                .map(Integer::parseInt)
+                .orElse(INTERNAL_HTTPS_PORT_DEFAULT);
+    }
+
+    public static URILike getInternalServiceHttpsUrl(ServiceContext ctx) {
+        var serviceName = ctx.getOwner().getName();
+        var projectName = ctx.<OpenShiftClient> get(CLIENT).project();
+        var host = "%s.%s.svc.cluster.local".formatted(serviceName, projectName);
+        return new URILike(Protocol.HTTPS.getValue(), host, getInternalHttpsPort(ctx), "");
+    }
+
+    private static PropertyToValue getPropertyToValue(String propertyValue, String prefix, String separator, String what) {
```

Comment:
```suggestion
    private static PropertyToValue getPropertyToValue(String propertyValue, String prefix, String separator, String subject) {
```
or something similar

Reply:
applied, yep, that's better